lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -1945,7 +1949,7 @@ bool SROAGlobalAndAllocas(HLModule &HLM, bool bHasDbgInfo) {
         continue;
       }
 
-      // Flat Global vector if no dynamic vector indexing.
+      // Flatten Global vector if no dynamic vector indexing.
```

Comment:
nit: If we're fixing a comment, might as well make it a complete sentence.
```suggestion
      // Flatten global vector if no dynamic vector indexing occurs.
```